tests/cephadm/test_daemon.py

```diff
@@ -26,16 +26,22 @@ def run(ceph_cluster, **kw):
           output, error   returned by the command.
     """
     config = kw.get("config")
-
-    # Manage Ceph using ceph-admin orchestration
-    command = config.pop("command")
-    log.info("Executing service %s service" % command)
-    daemon = Daemon(cluster=ceph_cluster, **config)
-
-    try:
-        method = fetch_method(daemon, command)
-        method(config)
-    finally:
-        # Get cluster state
-        get_cluster_state(daemon)
+    tests = config.get("steps") or [{"config": config}]
+
+    for test in tests:
+        # Manage Ceph daemon using ceph-admin daemon orchestration commands
+        _test_cfg = test["config"]
+        command = _test_cfg.pop("command")
+        log.info("Executing service %s service" % command)
+        daemon = Daemon(cluster=ceph_cluster, **_test_cfg)
+
+        try:
+            method = fetch_method(daemon, command)
+            method(_test_cfg)
+        except Exception as err:
+            log.error(err)
```

Comment:
Why need to catch exception and return `1`, even exception is raised this test case wiil fail

Reply:
yes, we can change this. since `run.py` handles this.

Reply:
In this case we don't need to try..exception here, but need to raise proper exception.